tests/e2e/e2e_util_test.go

```diff
@@ -102,27 +107,44 @@ func (s *IntegrationTestSuite) sendIBC(srcChainID, dstChainID, recipient string,
 		OutputStream: &outBuf,
 		ErrorStream:  &errBuf,
 	})
+
 	s.Require().NoErrorf(
 		err,
 		"failed to send IBC tokens; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
 	)
 
+	s.Require().Truef(
+		strings.Contains(outBuf.String(), "Success"),
+		"tx returned non code 0; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
+	)
+
+	currentHeight := s.chainHeight(s.valResources[srcChain.ChainMeta.Id][0].Container.ID)
+	// required to wait 3 blocks (~3 seconds) in order to prevent account sequence mismatches
+	s.Require().Eventually(
+		func() bool {
+			return s.chainHeight(s.valResources[srcChain.ChainMeta.Id][0].Container.ID) > currentHeight+2
+		},
+		5*time.Minute,
+		time.Second,
+	)
+
 	s.T().Log("successfully sent IBC tokens")
 }
 
-func (s *IntegrationTestSuite) submitProposal(c *chain.Chain) {
+func (s *IntegrationTestSuite) submitProposal(c *chain.Chain, upgradeHeight int) {
+	upgradeHeightStr := strconv.Itoa(upgradeHeight)
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	defer cancel()
 
-	s.T().Logf("submitting upgrade proposal for chain-id: %s", c.ChainMeta.Id)
+	s.T().Logf("submitting upgrade proposal on %s container: %s", s.valResources[c.ChainMeta.Id][0].Container.Name[1:], s.valResources[c.ChainMeta.Id][0].Container.ID)
```

Comment:
Why do we need to log both name and id?

Reply:
I like having both, the name is human readable and the container ID is used the check those containers logs and other docker functions. If you still would like me to remove one I can